packages/plugin-docusaurus/src/client/theme/SearchBar/index.tsx

```diff
@@ -118,7 +150,42 @@ export default function SearchBar(): JSX.Element {
     return () => {
       search.destroy()
     }
-  }, [isBrowser, siteConfig, indexData])
+  }, [isBrowser, siteConfig, database, colorMode])
+
+  useEffect(() => {
+    async function loadDatabase(): Promise<void> {
+      let data: RawDataWithPositions
+      if (!searchData) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const searchResponse = await fetch(searchBaseUrl.replace('@VERSION@', version!))
+
+        if (searchResponse.status === 0) {
+          throw new Error(`Network error: ${await searchResponse.text()}`)
```

Comment:
throwing errors in `useEffect` can cause very bad behavior. Should we put a generic error message for say "OOOPS something goes wrong" to the user?